backends/dropbox/index.js

```diff
@@ -18,17 +18,32 @@ export default class Dropbox extends OAuthBackend {
 	}
 
 	async upload (file, path) {
-		if (this.path) {
+		if (this.ref.path) {
+			path = this.ref.path.replace(/[^/]+$/, "") + path; // make upload path relative to existing path
+		}
+		else {
 			path = path.startsWith("/") ? path : "/" + path;
-			path = this.path + path;
 		}
 
-		await this.put(file, {path});
+		await this.put(file, {ref: path, isUploading: true});
```

Comment:
I thought it was `file` that we renamed to `ref`?

Reply:
It's inside the `upload()` method where we get a `File` object as input the way it's [described in our docs](https://madata.dev/#uploads):

```html
<input type=file id=uploader />
```

```js
uploadForm.addEventListener("submit", evt => {
	evt.preventDefault();

	let file = uploader.files[0];

	if (file && file.type.startsWith("image/")) {
		backend.upload(file, `images/${file.name}`);
	}
});
```

So, it's not the `ref` object.

We have the same thing in our another file-based backends:

https://github.com/madatajs/madata/blob/0a0f1ac93adf0b31cc63bc8b62fd7d83a992163d/backends/github/file/github-file.js#L197

https://github.com/madatajs/madata/blob/0a0f1ac93adf0b31cc63bc8b62fd7d83a992163d/backends/firebase/index.js#L187

https://github.com/madatajs/madata/blob/0a0f1ac93adf0b31cc63bc8b62fd7d83a992163d/backends/google/drive/google-drive.js#L128

Reply:
Oh I see. Still, it doesn't seem like we should rename `path` to `ref` here? `ref` should not be used to mean anything other than "object with params about the backend location".

Reply:
Oh. Right. We need to get `ref` from the relative URL with [`Backend._getRef(path)`](https://github.com/madatajs/madata/blob/0a0f1ac93adf0b31cc63bc8b62fd7d83a992163d/src/backend.js#L160-L174) then. Let me push the fix. Thanks!

Reply:
Done!